### ISSUE TITLE
Corrupted kdbx file

### ISSUE THREAD

**Issue by skoam** [keeweb/keeweb#760]
Hey,

I am opening this ticket and I hope we can do something about this. I have been using Keeweb for 2 years now and today it corrupted my database to a point where not even the Import (Repair) function of Keepass2 produces anything. Linux interprets the corrupted files as "STL 3D Model" instead of kdbx files now.

I am now looking for opportunities to search for what happened, but I cannot find anything that would do so. Are there any log files or backup caches keeweb creates to prevent this from happening?

Here is what I did:

- I am storing the database on a partition formatted in Ext4, which I made available in both Windows and Linux. So far there have been no issues with that.
- Today I opened my database and was looking for some entries. While I did so, I noticed a strange entry at the top of the list on the right, between tags and my database name: An entry which consisted mostly of numbers and letters. It contained one entry which I already had in the database, but I wanted to double check if I actually have that entry. (I have now noticed that this entry was a tag, probably created by accident from a password)
- With that, I copied the password and went to the designated entry in my database
- To compare the passwords, I tried to temporarily copy the password to a visible field
- I first copied it to the expiration date by accident, the field was emptied automatically
- After that I copied the password to "more.." and compared it to the real password. They matched, so I removed it from "more.."

After that, I saved the database. I think the corruption already happened there, but the database was still open.

I only noticed that it was broken after I renamed a folder, gave it an icon and dragged some entries from a different folder into it. After that, I noticed that I wasn't able to save my file any longer (it showed an orange dot in the lower left corner).

I went to storing options and tried to rename it, save it as a different file or do anything, nothing would save the file.

I made a backup copy before I attempted the drag & drop to move entries around, as I was afraid that this would be more severe. However, the backed up copy already was corrupted and nothing will open it anymore.

I'll try around an check if I can reproduce this, but I am currently out of luck.

This really sucks. My last backup is some months old and this shouldn't happen without there being some kind of fallback or warning. At the point where the saving failed, the file was still open in keeweb. It should've offered me to repair it, save the entries to an XML or *something*.

I'll update this if I find something reproducable.

**Comment by skoam** [comment]
...

wow.

Broken in HEX editor:
https://i.imgur.com/1nHFrlg.png

Backup from months ago:
https://i.imgur.com/QwtgGAc.png

I expected some 00's at the top but.. wow.

**Comment by skoam** [comment]
I also tried renaming the file while it was still open, creating a copy of my database. It appears in the list of available databases, but opening it yields:

Error: ENOENT: no such file or directory, open 'C:\Users\user\AppData\Roaming\KeeWeb\OfflineFiles\565369cf-6769-7021-602a-c545f03d3a64'

**Comment by skoam** [comment]
Following the steps above is not reproducing the issue. I'm out of ideas.

**Comment by antelle** [comment]
Hi,
It looks like there was a weird entry and the file got corrupted, however I still don't understand how it happened. It would be very helpful if you could reproduce it. There is an option to export xml, in file settings.
Did you have any attachments in the db?
Was it synced to Dropbox, GDrive, etc...?

**Comment by skoam** [comment]
No syncing, just the plain file on an ext4 partition, made available by a special driver to make windows detect it. However, that seems not to be an issue as that would probably cause issues immediately and repeatedly.

The file has no attachments as far as I know.

I also saw the option to export it to XML in the settings (I think I saw the Bad Signature error also appearing on that page) but I didn't jump to the right conclusion fast enough and decided to close it before I noticed that even my backup was already broken.

I'll give reproducing it a couple more shots tomorrow.
